File: snail-level-4-kata.py
def snail(array):
  result=[]
  while (len(array)):
    if(result):
#gets top row 
        result+=array.pop(0) 
    else:
#gets first top row
        result=array.pop(0)
#gets far right column
    for i in range(len(array)):
      result.append(array[i].pop())
    if(array):
#gets bottom row and reverses it
        result += reversed(array.pop())
#gets far left column
    for i in range(1,len(array)):
      result.append(array[len(array)-i].pop(0))
  return result;

File: snail-level-4-kata.js
snail = function(array) {
    var result;
  while (array.length) {
    // gets the first row.
    result = (result ? result.concat(array.shift()) : array.shift());
    // gets the right column.
    for (var i = 0; i < array.length; i++)
      result.push(array[i].pop());
    // gets the bottom row.
    result = result.concat((array.pop() || []).reverse());
    // gets the left column.
    for (var i = array.length - 1; i >= 0; i--)
      result.push(array[i].shift());
  }
  return result;
}